test-snapshots/snapshots-input/Snapshot.Branch.purs

```diff
@@ -0,0 +1,23 @@
+module Snapshot.Branch where
+
+f :: Boolean -> Boolean -> Boolean -> Int
+f x y z = if x then if y then if z then 0 else 1 else 2 else 3
+
+g :: Int -> Int
+g = case _ of
+  0 -> 1
+  1 -> 2
+  2 -> 3
+  _ -> 0
+
+h :: Number -> Number
+h = case _ of
+  3.14 -> 3.14159
+  _ -> 0.0
```

Comment:
Since you now convert `Fail`, would using a partial pattern match work so we can verify the `Fail` is being outputted correctly?
```
fail :: Partial => Int -> Int
fail = unsafePartial case _ of
  1 -> 1
```

If not, perhaps we shouldn't convert it until ADTs are implemented.